pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationManager.java

```diff
@@ -177,6 +190,29 @@ private boolean checkCluster(DestinationName destination) {
         return permissionFuture;
     }
 
+    private Boolean checkWildcardPermission(String checkedRole, AuthAction checkedAction,
+            Map<String, Set<AuthAction>> permissionMap) {
+        for (Map.Entry<String, Set<AuthAction>> permissionData : permissionMap.entrySet()) {
+            String permittedRole = permissionData.getKey();
+            Set<AuthAction> permittedActions = permissionData.getValue();
+
+            // Prefix match
+            if (permittedRole.charAt(permittedRole.length() - 1) == '*'
+                    && checkedRole.startsWith(permittedRole.substring(0, permittedRole.length() - 1))
+                    && permittedActions.contains(checkedAction)) {
+                return true;
+            }
+
+            // Suffix match
+            if (permittedRole.charAt(0) == '*'
```

Comment:
as we just check Prefix and  suffix so, it seems it will not support wild-card character for usecase:
checkedRole = `abcd` 
and permittedRole = `ab*d`
So, instead can we use regex-pattern to validate it.